apps/judicial-system/web/src/routes/Prosecutor/InvestigationRequest/Defendant/Defendant.tsx

```diff
@@ -31,48 +35,52 @@ const Defendant = () => {
   }, [])
 
   const handleNextButtonClick = async (theCase: Case) => {
-    if (!theCase.id) {
-      const createdCase = await createCase(theCase)
+    try {
+      if (!theCase.id) {
+        const createdCase = await createCase(theCase)
 
-      if (createdCase) {
-        workingCase.defendants?.forEach(async (defendant, index) => {
-          if (
-            index === 0 &&
-            createdCase.defendants &&
-            createdCase.defendants.length > 0
-          ) {
-            await updateDefendant(
-              createdCase.id,
-              createdCase.defendants[0].id,
-              {
+        if (createdCase) {
+          workingCase.defendants?.forEach(async (defendant, index) => {
+            if (
+              index === 0 &&
+              createdCase.defendants &&
+              createdCase.defendants.length > 0
+            ) {
+              await updateDefendant(
+                createdCase.id,
+                createdCase.defendants[0].id,
+                {
+                  gender: defendant.gender,
+                  name: defendant.name,
+                  address: defendant.address,
+                  nationalId: defendant.nationalId,
+                  noNationalId: defendant.noNationalId,
+                  citizenship: defendant.citizenship,
+                },
+              )
+            } else {
+              await createDefendant(createdCase.id, {
                 gender: defendant.gender,
                 name: defendant.name,
                 address: defendant.address,
                 nationalId: defendant.nationalId,
                 noNationalId: defendant.noNationalId,
                 citizenship: defendant.citizenship,
-              },
-            )
-          } else {
-            await createDefendant(createdCase.id, {
-              gender: defendant.gender,
-              name: defendant.name,
-              address: defendant.address,
-              nationalId: defendant.nationalId,
-              noNationalId: defendant.noNationalId,
-              citizenship: defendant.citizenship,
-            })
-          }
-        })
-        router.push(
-          `${constants.IC_HEARING_ARRANGEMENTS_ROUTE}/${createdCase.id}`,
-        )
+              })
+            }
+          })
+          router.push(
+            `${constants.IC_HEARING_ARRANGEMENTS_ROUTE}/${createdCase.id}`,
+          )
+        } else {
+          // TODO handle error
+          return
+        }
       } else {
-        // TODO handle error
-        return
+        router.push(`${constants.IC_HEARING_ARRANGEMENTS_ROUTE}/${theCase.id}`)
       }
-    } else {
-      router.push(`${constants.IC_HEARING_ARRANGEMENTS_ROUTE}/${theCase.id}`)
+    } catch (error) {
+      toast.error(formatMessage(errors.general))
```

Comment:
The method `createCase` in the `useCase` hook has its own error handling. Consider adding similar error handling to the methods `updateDefendant` and `createDefendant` in the `useDefendant` hook and stop catching errors here.